.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,107 @@
+name: benchmark
+
+on: workflow_call # TODO: add pull_request-labeled with "benchmark" label when specifying a label is supported
+
+jobs:
+  start-runner:
+    name: Start self-hosted EC2 runner
+    runs-on: ubuntu-latest
+    outputs:
+      label: ${{ steps.start-ec2-runner.outputs.label }}
+      ec2-instance-id: ${{ steps.start-ec2-runner.outputs.ec2-instance-id }}
+    steps:
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: ${{ secrets.AWS_REGION }}
+      - name: Start EC2 runner
+        id: start-ec2-runner
+        uses: machulav/ec2-github-runner@v2
+        with:
+          mode: start
+          github-token: ${{ secrets.GH_PERSONAL_ACCESS_TOKEN }}
+          # Ubuntu 22.04 128GB Storage AMI
+          ec2-image-id: ami-0ba430d4b7b64de57
+          ec2-instance-type: r6i.large
+          subnet-id: ${{ secrets.AWS_EC2_SUBNET_ID }}
+          security-group-id: ${{ secrets.AWS_EC2_SG_ID }}
+
+  benchmark:
+    name: Run the benchmarks on the runner
+    needs: start-runner # required to start the main job when the runner is ready
+    runs-on: ${{ needs.start-runner.outputs.label }} # run the job on the newly created runner
+    steps:
+      - name: checkout
+        uses: actions/checkout@v3
+
+      - name: Print runner info
+        run: |
+          printf "Runner lscpu:\n$(lscpu)\n"
+          printf "Runner lsmem:\n$(lsmem)\n"
+          printf "Runner nproc:\n$(nproc)\n"
+          printf "Runner uname:\n$(uname -a)\n"
+
+      - name: Install benchmark dependencies
+        run: |
+             sudo .install/install_script.sh
+             sudo apt install python3-pip -y
+             pip3 install --upgrade pip PyYAML setuptools
```

Comment:
Any reason not to combine the pip lines... or even have your install script do this?